src/main/java/org/wso2/carbon/module/swiftiso20022/JsonToMT103Validator.java

```diff
@@ -0,0 +1,125 @@
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+import org.wso2.carbon.module.swiftiso20022.constants.MT103Constants;
+import org.wso2.carbon.module.swiftiso20022.models.mt103models.MT103JsonRequestPayload;
+import org.wso2.carbon.module.swiftiso20022.utils.ConnectorUtils;
+import org.wso2.carbon.module.swiftiso20022.utils.JsonToMt103Utils;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidationResult;
+
+import java.util.Optional;
+
+
+/**
+ * Class to convert JSON Object to MT103 format.
+ */
+public class JsonToMT103Validator extends AbstractConnector {
+
+    private static final Gson gson = new Gson();
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+
+        this.log.debug("Executing JsonToMT103Transformer to convert the JSON payload to MT103 format");
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+
+        // get JSON payload as a string from the message context
+        Optional<String> payload = ConnectorUtils.buildMessagePayloadFromMessageContext(axis2MessageContext);
+
+        // if JSON payload is available
+        if (payload.isPresent()) {
+
+            // check JSON object with request model (data types of fields)
+            try {
+                validatePayloadWithMT103JsonModel(payload.get());
+            } catch (JsonSyntaxException e) {
+                this.log.error(ConnectorConstants.ERROR_INVALID_PAYLOAD, e);
+                ConnectorUtils.appendErrorToMessageContext(
+                        messageContext, ConnectorConstants.INVALID_REQUEST_PAYLOAD, e.getMessage());
+                super.handleException(ConnectorConstants.ERROR_INVALID_PAYLOAD, messageContext);
+            }
+
+            // validate values of the JSON object
+            this.log.debug("Validating payload parameters");
+            ValidationResult validationResponse = validateRequestPayload(payload.get());
+            if (validationResponse.isInvalid()) {
+                this.log.error(validationResponse.getErrorMessage());
+                ConnectorUtils.appendErrorToMessageContext(messageContext, validationResponse.getErrorCode(),
+                        validationResponse.getErrorMessage());
+                super.handleException(ConnectorConstants.ERROR_VALIDATION_FAILED, messageContext);
+            }
+        } else {
+            this.log.error(ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            ConnectorUtils.appendErrorToMessageContext(messageContext, ConnectorConstants.MISSING_REQUEST_PAYLOAD,
+                    ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            super.handleException(ConnectorConstants.ERROR_MISSING_PAYLOAD, messageContext);
+        }
+    }
+
+    private ValidationResult validateRequestPayload(String payloadString) {
```

Comment:
Shall we add method comment

Reply:
Added in commit 96d4dc498094daee16277501378ee52fbb484dd1